cleanlab/regression/rank.py

```diff
@@ -181,12 +180,7 @@ def _get_outre_score_for_each_label(
     features = np.array([labels, residual]).T
 
     neighbors = int(np.ceil(frac_neighbors * labels.shape[0]))
-    if neighbor_metric is None:
-        if features.shape[0] > 100:
-            neighbor_metric = "euclidean"
-        else:
-            neighbor_metric = euclidean
-    knn = NearestNeighbors(n_neighbors=neighbors, metric=neighbor_metric).fit(features)
+    knn = features_to_knn(features, n_neighbors=neighbors, metric=neighbor_metric)
```

Comment:
Because the `features` are 2d, then we can safely assume that `features_to_knn` will pick a suitable metric, even if `neighbor_metric = None`.